### PR TITLE
Hotfix to make particles compile without MPI again

### PR DESCRIPTION
**Pull request by pgrete**
## PR Summary

Fixing non-MPI builds with particles and HostCommBuffer support.

Note, that the removed for loop was present twice back-to-back in those function (if I didn't miss anything)

## PR Checklist

- [x] Code passes cpplint
- [ ] New features are documented.
- [x] Adds a test for any bugs fixed. Adds tests for new features.
- [x] Code is formatted
- [x] Changes are summarized in CHANGELOG.md
- [ ] CI has been triggered on [Darwin](https://re-git.lanl.gov/eap-oss/parthenon/-/pipelines) for performance regression tests.
- [ ] (@lanl.gov employees) Update copyright on changed files

### CODE REVIEW

File: example/particle_leapfrog/particle_leapfrog.cpp [hunk 1]
```diff
@@ -394,17 +395,7 @@ TaskStatus InitializeCommunicationMesh(const BlockList_t &blocks) {
...
-  for (auto &block : blocks) {
-    auto &pmb = block;
-    auto sc = pmb->swarm_data.Get();
-    auto swarm = sc->Get("my particles");
-
-    for (int n = 0; n < swarm->vbswarm->bd_var_.nbmax; n++) {
-      auto &nb = pmb->pbval->neighbor[n];
-      swarm->vbswarm->bd_var_.flag[nb.bufid] = BoundaryStatus::waiting;
-    }
-  }
```
Comment (Yurlungur) [inline example/particle_leapfrog/particle_leapfrog.cpp:408]: I assume this isn't necessary anymore?

**Review by Yurlungur — APPROVED** [review]
Good catch!

File: example/particle_leapfrog/particle_leapfrog.cpp (hunk 1 above)
Comment (pgrete) [reply example/particle_leapfrog/particle_leapfrog.cpp:408]: That the loop that is duplicated. If I didn't miss anything the exact same loop follows this one. Probably a copy/paste leftover.

**Comment by pgrete** [comment]
FYI I just triggered the full test suite.

**Comment by pgrete** [comment]
Full pipeline passed. One more approval and this fix can go in,

File: example/particle_leapfrog/particle_leapfrog.cpp (hunk 1 above)
Comment (brryan) [reply example/particle_leapfrog/particle_leapfrog.cpp:408]: Yep just looks like a mistaken duplication

**Comment by brryan** [comment]
Nice fix thanks @pgrete

**Review by brryan — APPROVED** [review]

**Pull request merged**